src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd

```diff
@@ -531,9 +531,10 @@
 
     <xsd:complexType name="lock">
         <xsd:sequence>
-            <xsd:element name="resource" type="lock_resource" minOccurs="1" maxOccurs="unbounded" />
+            <xsd:element name="resource" type="lock_resource" minOccurs="0" maxOccurs="unbounded" />
         </xsd:sequence>
         <xsd:attribute name="enabled" type="xsd:boolean" />
+        <xsd:attribute name="resource" type="xsd:string" />
```

Comment:
why allowing a resource attribute here ? this does not make to me: we still need a name for the lock being registered, so `<framework:resource name="my_lock">my_service</framework:resource>` is still the expected XML format to me.

Reply:
same for semaphore of course

Reply:
@stof This is the equivalent of doing:

```yaml
framework:
    lock: dsn-or-service
```

If it's a string, the `default` name is used:

https://github.com/symfony/symfony/blob/4e70834e5d3f2a42f041dd2d1c89fd142ea61ece/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1392-L1394

Reply:
The issue is that the XML schema allows providing *both* the attribute and child elements, which would do weird stuff. I'd rather keep the clean XML schema as we have right now, even though it does not support one of the shortcut notation.

If someone is using XML configuration file, they will not complain to us that their configuration is too verbose due to that missing shortcut, as using XML in itself would be responsible for a lot more verbosity. Projects using XML configuration file generally do it to benefit from the XML schema.

Reply:
@stof Makes sense, see https://github.com/symfony/symfony/compare/c2a12b45832d1262ddb2e0ea9ccf545bef91fc74..34e5112e9afaf2ef4a9fb348a90b8bfeae71c6db